### ISSUE TITLE
Accuracy and IoU is decreasing

### ISSUE THREAD

**Issue by TolgaBkm** [open-mmlab/mmsegmentation#2069]
Hello,
I'm training Segformer for a binary segmentation for glaciers. In my annotations, background is coded as 0, and glacier is coded as 1.
I'm doing a experimental training with 4000 iteration steps and performing evaluation on each 200 steps.
The training starts up great in the first 200 step:
![image](https://user-images.githubusercontent.com/57003251/190158163-eb7c0aec-6b28-4cac-8efc-af7967850599.png)
However, as the training goes on, the values starts to decrease dramatically. Here are 400th step values:
![image](https://user-images.githubusercontent.com/57003251/190158267-46b4db12-8d10-4ee8-8d9a-eb46634311dc.png)
600:
![image](https://user-images.githubusercontent.com/57003251/190158327-5fc7604b-971d-4df2-a1d9-b906c19abc98.png)
1800:
![image](https://user-images.githubusercontent.com/57003251/190158559-44d638d9-c2e4-4243-ba3b-a9bd9ca43f77.png)
In the end, the trained model predicts whole images as glacier (1).

Is there something I'm doing wrong?
Here is my config:
`norm_cfg = dict(type='BN', requires_grad=True)
model = dict(
    type='EncoderDecoder',
    pretrained=
    'https://download.openmmlab.com/mmsegmentation/v0.5/pretrain/segformer/mit_b5_20220624-658746d9.pth',
    backbone=dict(
        type='MixVisionTransformer',
        in_channels=3,
        embed_dims=64,
        num_stages=4,
        num_layers=[3, 6, 40, 3],
        num_heads=[1, 2, 5, 8],
        patch_sizes=[7, 3, 3, 3],
        sr_ratios=[8, 4, 2, 1],
        out_indices=(0, 1, 2, 3),
        mlp_ratio=4,
        qkv_bias=True,
        drop_rate=0.0,
        attn_drop_rate=0.0,
        drop_path_rate=0.1),
    decode_head=dict(
        type='SegformerHead',
        in_channels=[64, 128, 320, 512],
        in_index=[0, 1, 2, 3],
        channels=256,
        dropout_ratio=0.1,
        num_classes=2,
        norm_cfg=dict(type='BN', requires_grad=True),
        align_corners=False,
        loss_decode=dict(
            type='CrossEntropyLoss', use_sigmoid=False, loss_weight=1.0)),
    train_cfg=dict(),
    test_cfg=dict(mode='whole'))
dataset_type = 'GlacierDataset'
data_root = 'data/GlacierDataset'
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_rgb=True)
crop_size = (512, 512)
train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='LoadAnnotations', reduce_zero_label=False),
    dict(type='RandomFlip', prob=0.5),
    dict(type='PhotoMetricDistortion'),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_rgb=True),
    dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
    dict(type='DefaultFormatBundle'),
    dict(type='Collect', keys=['img', 'gt_semantic_seg'])
]
test_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(
        type='MultiScaleFlipAug',
        img_scale=(2048, 512),
        flip=False,
        transforms=[
            dict(type='Resize', keep_ratio=True),
            dict(type='RandomFlip'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='ImageToTensor', keys=['img']),
            dict(type='Collect', keys=['img'])
        ])
]
data = dict(
    samples_per_gpu=2,
    workers_per_gpu=0,
    train=dict(
        type='GlacierDataset',
        data_root='data/GlacierDataset',
        img_dir='images/training',
        ann_dir='annotations/training',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(type='LoadAnnotations', reduce_zero_label=False),
            dict(type='RandomFlip', prob=0.5),
            dict(type='PhotoMetricDistortion'),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_rgb=True),
            dict(type='Pad', size=(512, 512), pad_val=0, seg_pad_val=255),
            dict(type='DefaultFormatBundle'),
            dict(type='Collect', keys=['img', 'gt_semantic_seg'])
        ]),
    val=dict(
        type='GlacierDataset',
        data_root='data/GlacierDataset',
        img_dir='images/validation',
        ann_dir='annotations/validation',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(512, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]),
    test=dict(
        type='GlacierDataset',
        data_root='data/GlacierDataset',
        img_dir='images/test',
        ann_dir='annotations/test',
        pipeline=[
            dict(type='LoadImageFromFile'),
            dict(
                type='MultiScaleFlipAug',
                img_scale=(512, 512),
                flip=False,
                transforms=[
                    dict(type='Resize', keep_ratio=True),
                    dict(type='RandomFlip'),
                    dict(
                        type='Normalize',
                        mean=[123.675, 116.28, 103.53],
                        std=[58.395, 57.12, 57.375],
                        to_rgb=True),
                    dict(type='ImageToTensor', keys=['img']),
                    dict(type='Collect', keys=['img'])
                ])
        ]))
log_config = dict(
    interval=50, hooks=[dict(type='TextLoggerHook', by_epoch=False)])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = 'checkpoints/segformer_mit-b5_512x512_160k_ade20k_20210726_145235-94cedf59.pth'
resume_from = None
workflow = [('train', 1)]
cudnn_benchmark = True
optimizer = dict(
    type='AdamW',
    lr=6e-05,
    betas=(0.9, 0.999),
    weight_decay=0.01,
    paramwise_cfg=dict(
        custom_keys=dict(
            pos_block=dict(decay_mult=0.0),
            norm=dict(decay_mult=0.0),
            head=dict(lr_mult=10.0))))
optimizer_config = dict()
lr_config = dict(
    policy='poly',
    warmup='linear',
    warmup_iters=1500,
    warmup_ratio=1e-06,
    power=1.0,
    min_lr=0.0,
    by_epoch=False)
runner = dict(type='IterBasedRunner', max_iters=4000)
checkpoint_config = dict(by_epoch=False, interval=200)
evaluation = dict(interval=200, metric='mIoU', pre_eval=True)
checkpoint = 'https://download.openmmlab.com/mmsegmentation/v0.5/pretrain/segformer/mit_b5_20220624-658746d9.pth'`

**Comment by MeowZheng** [comment]
I think the num of pixels for each category might be imbalanced, my suggestion is to check the ground truth, if it is imbalanced, you might try some training tricks to solve the problem.

**Comment by androbaza** [comment]
I've had similar problem with imbalanced classes, and  FocalLoss has helped me to overcome it.

**Comment by chefkrym** [comment]
> First, I tried Focal Loss as @androbaza suggested. However, the results were actually worse. Then, I created a balanced dataset as @MeowZheng advised. The results were better but still not satisfactory. Finally, I decreased learning rate and warm up ratio. This seems to kind of resolve my issue. I get the best results on validation set around 3.6 k iterations.

Hey @TolgaBkm @androbaza , I'm totally new and wanted to do a binary segmentation on my own dataset using Segformer. Could you be so kind as to share your working code please? Thank you so much :)

**Comment by TolgaBkm** [comment]
The attachment includes the notebook and py file that I have used.
Please note my code uses "tif" files. You should change the image suffix in the notebook according to the file type you are using.
[knet_test.zip](https://github.com/open-mmlab/mmsegmentation/files/11541324/knet_test.zip)

**Comment by chefkrym** [comment]
> The attachment includes the notebook and py file that I have used. Please note my code uses "tif" files. You should change the image suffix in the notebook according to the file type you are using. [knet_test.zip](https://github.com/open-mmlab/mmsegmentation/files/11541324/knet_test.zip)

@TolgaBkm Thank you SO SO much, you're a life saver!
